tests/utils/verification/dockerhost.go

```diff
@@ -0,0 +1,100 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This util provides various helper methods that can be used by different tests to
+// do verification on docker host (VM)
+
+package verification
+
+import (
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/vmware/docker-volume-vsphere/tests/constants/vm"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/misc"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/ssh"
+)
+
+// CheckVolumeInProcMounts - check if volume entry is present in docker host mount file
+// grep does the work. If entry isn't present, err is not nil (Exit status 1)
+func CheckVolumeInProcMounts(hostName, volName string) bool {
+	cmd := "grep -q " + volName + vm.MountFile
+	_, err := ssh.InvokeCommand(hostName, cmd)
+
+	if err != nil {
+		log.Printf("No entry for volume %s in %s.", volName, vm.MountFile)
+		return false
+	}
+
+	return true
+}
+
+// CheckRefCount - check the log record for expected refcount of the volume.
+// grep does the work. If refcount isn't the expected one, err is not nil (Exit status 1)
+// loglogLines are number of log lines to be tailed from the log file.
+// eg: tail -1 /var/log/docker-volume-vsphere.log | grep testName |  grep -q refcount=5
+func CheckRefCount(hostName, volName string, logLines, expectedRefCnt int) bool {
+	cmd := "tail -" + strconv.Itoa(logLines) + vm.LogFile + " | grep " + volName + " | grep -q refcount=" +
+		strconv.Itoa(expectedRefCnt)
+	_, err := ssh.InvokeCommand(hostName, cmd)
+
+	if err != nil {
+		log.Printf("Unable to find refcount for %s in logs.", volName)
+		return false
+	}
+
+	return true
+}
+
+// CheckRecoveryRecord - check the recovery record printed by plugin after restart
+// The log record must contain expected refcount
+// loglogLines are number of log lines to be tailed from the log file.
+// eg: tail -50 /var/log/docker-volume-vsphere.log | grep 'Volume name=testName' | grep 'mounted=true'
+func CheckRecoveryRecord(hostName, volName string, logLines, expectedRefCnt int) bool {
+	cmd := "tail -" + strconv.Itoa(logLines) + vm.LogFile + " | grep 'Volume name=" + volName +
+		"' | grep 'mounted=true'"
+	op, err := ssh.InvokeCommand(hostName, cmd)
+	expectedStr := "count=" + strconv.Itoa(expectedRefCnt)
+	if err != nil {
+		log.Printf("No recovery record for %s in logs.", volName)
+		return false
+	}
+
+	log.Printf("Recovery record is %s ", op)
+
+	if strings.Contains(op, expectedStr) != true {
+		log.Printf("Recovery record %s doesn't match expected refcount %d", op, expectedRefCnt)
+		return false
+	}
+	return true
+}
+
+// VerifyRecoveryRecord Verify the recovery record. There is a delay after which recovery record appears in the log
+// due to refcounting being delayed in background. (Docker doesn't respond immediately)
+func VerifyRecoveryRecord(hostName, volName string, logLines, expectedRefCnt int) bool {
```

Comment:
It will be confusing to expose two functions with similar names: VerifyRecoveryRecord and CheckRecoveryRecord. Looks like the latter is being invoked by the former only, so I think we can make CheckRecoveryRecord unexported.